### DISCUSSION TITLE
make option to display / show all debug options?

### DISCUSSION THREAD

**Discussion by berlin4apk** [ipxe/ipxe#375] [Q&A] [answered]
Is there a way to display all debug options?
I found in the discussion this one:
```
debug="device,runtime,pci,init,nii,snp,snpnet,snponly,efi_watchdog,efi_image,image_cmd,image,efi_autoboot,efi_timer,efi_driver:3,efi_debug,efi_init,efi_pci,efi_wrap,intel,intelxl"
```

But I was not able to find it on the wiki.

**Comment by mcb30** [comment, marked as answer]
https://ipxe.org/download#debug_builds has the relevant documentation

**Reply by berlin4apk** [reply]
Ok thank you for the link,
But I don't understand how / where I find the "debug keywords" 
e.g. I want to debug an ipxe script
As I have a ipxe.iso with bios / EFI / 32bit / 64bit support.
But it failed if I included a embedded ipxe script

Working is like:
```
make bin-i386-efi/ipxe.efi bin-x86_64-efi/ipxe.efi bin/ipxe.lkrn
./util/genfsimg -o my.iso bin-i386-efi/ipxe.efi bin-x86_64-efi/ipxe.efi bin/ipxe.lkrn
```
But With a specific ipxe script form me it doesn't work,
Iso build via
```
make bin-i386-efi/ipxe.efi bin-x86_64-efi/ipxe.efi bin/ipxe.lkrn
./util/genfsimg -o my.iso -s script.ipxe bin-i386-efi/ipxe.efi bin-x86_64-efi/ipxe.efi bin/ipxe.lkrn
```

**Reply by berlin4apk** [reply]
After I can pin point the problem in the script, I will open a detailed issue

**Reply by mcb30** [reply]
> Ok thank you for the link,
> But I don't understand how / where I find the "debug keywords"

As documented at the link I provided, the "debug keywords" are just the names of the source files (without the `.c` suffix)

**Reply by NiKiZe** [reply]
"For example, to debug problems involving iSCSI, it may be useful to enable debugging of the iSCSI protocol driver (in iscsi.c):" here the .c is kind of explained. Debug can be used for any .c file that contains DBG statements. So grep -r DBG

**Comment by berlin4apk** [comment]
> > Ok thank you for the link,
> > But I don't understand how / where I find the "debug keywords"
> 
> As documented at the link I provided, the "debug keywords" are just the names of the source files (without the `.c` suffix)

THIS was the missing link for me!
Thank you for the hint.

Can we add this to the wiki?
As a PR for the wiki content is until now not possible if I'm not mistaken
